avm/res/container-service/managed-cluster/main.bicep

```diff
@@ -622,6 +631,11 @@ resource managedCluster 'Microsoft.ContainerService/managedClusters@2024-03-02-p
               any(dnsZoneResourceId)
             ]
           : null
+        nginx: !empty(defaultIngressControllerType)
+          ? {
+              defaultIngressControllerType: any(defaultIngressControllerType)
+            }
+          : null
```

Comment:
Given the error message in the deployment test it 'may' make sense to refactor this to
 ```bicep
...(!empty(defaultIngressControllerType)
  ? {
      nginx: {
        defaultIngressControllerType: any(defaultIngressControllerType)
      }
    }
  : {})
```
which only adds the property `nginx` to the parameter block if its not empty - as opposed to what's happening right now, that is, sets it with a `null` value if empty.
I think it's worth testing.

Reply:
the feature is only availeble with kubernetes version >=1.30, in the test file it was an older version, that was the issue.

test is now green!